src/AutoHashEqualsCached.jl

```diff
@@ -0,0 +1,212 @@
+module AutoHashEqualsCached
+
+export @auto_hash_equals_cached, @auto_hash_equals
+
+"""
+  `_show_default_auto_hash_equals_cached` is just like `Base._show_default(io, x)`,
+  except it ignores fields named `_cached_hash`.  This function is called in the
+  implementation of `T._show_default` for types `T` annotated with
+  `@auto_hash_equals_cached`.  This is ultimately used in the implementation of
+  `Base.show`.  This specialization ensures that showing circular data structures does not
+  result in infinite recursion.
+"""
```

Comment:
Take a look at https://github.com/RelationalAI/RAIStyle#documentation for info on formatting docstrings.

If this is indeed private, I'd change it to a comment block instead.

It's also a little atypical to see a docstring on a method prefixed with an `_` as that indicates it's internal and not part of the "public API", while a docstring tends to indicate it is part of the "public API".

Note: Julia doesn't really have public/private API's, so this contract is _very_ loose, but tends to come into play when determining whether something is a breaking change for semver reasons.

Reply:
Done.